### PR TITLE
[RFR] Initial framework commit for playwright tests

### PR DESCRIPTION
**Pull request by sshveta**

### CODE REVIEW

File: scripts/populate-konveyor.sh [hunk 1]
```diff
@@ -44,12 +44,10 @@ tmp=/tmp/${self}-${pid}
...
 usage() {
   echo "Usage: ${self} <required> <options>"
-  echo "-h help"
   echo "Auth has to be disabled in order to execute this script"
+  echo "-h help"
   echo "Required:"
-  echo "  -u <URL> URL of the hub"
-  echo "Example:"
-  echo "  ${self} -u htps://my-konveyor-instance/hub -n 5"
+  echo "  -u <URL>/hub"
```
Comment (abrugaro) [inline scripts/populate-konveyor.sh:50]: Why are we removing the example?

File: playwright.config.ts [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    {
+      name: 'firefox',
+      use: { ...devices['Desktop Firefox'] },
+    },
+
+    {
+      name: 'webkit',
+      use: { ...devices['Desktop Safari'] },
+    },
```
Comment (abrugaro) [inline playwright.config.ts:50]: We do not need this as we're going to test an electron app

File: e2e/pages/launch_vscode.page.js
Comment (abrugaro) [inline e2e/pages/launch_vscode.page.js:1]: We should be using `ts` instead of `js` files.

File: e2e/pages/launch_vscode.page.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
+// launch_vscode.page.js
+const { _electron: electron } = require('playwright');
+const path = require('path');
+const fs = require('fs');
+
+class LaunchVSCodePage {
```
Comment (abrugaro) [inline e2e/pages/launch_vscode.page.js:6]: If we're going to use this file as the main class for handling the interaction with vscode, maybe we can name the class just "VSCode".

File: e2e/pages/launch_vscode.page.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    this.vscodeApp = null;
+    this.window = null;
+  }
+
+  async launchVSCode(executablePath) {
+    this.vscodeApp = await electron.launch({
+      executablePath: executablePath, // Path to VSCode executable
+    });
+
+  }
```
Comment (abrugaro) [inline e2e/pages/launch_vscode.page.js:17]: Since an instance cannot be constructed without actually opening vscode, we could just make "launchVScode" static and return a class instance from there.

File: e2e/tests/vscode.test.ts [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+const path = require('path');
+
+test.describe('VSCode Extension Installation from VSIX', () => {
+   test('should install extension from a VSIX file in VSCode', async ({page}) => {
+    
+    // Path to the VSCode executable
+    const vscodeExecutablePath = '/usr/bin/code';  // Change this to the actual path
+
+    // Path to the VSIX file
+    const vsixFilePath = path.resolve(__dirname, '/home/sshveta/Downloads/kai-vscode-plugin-0.0.3.vsix');  // Provide the path to your VSIX file
```
Comment (abrugaro) [inline e2e/tests/vscode.test.ts:13]: TODO: move to env variables

**Review by abrugaro — CHANGES_REQUESTED** [review]

File: e2e/pages/launch_vscode.page.js [hunk 6]
```diff
@@ -0,0 +1,45 @@
+// launch_vscode.page.js
+const { _electron: electron } = require('playwright');
+const path = require('path');
+const fs = require('fs');
```
Comment (abrugaro) [inline e2e/pages/launch_vscode.page.js:4]: Replace with `import`s

File: e2e/tests/vscode.test.ts [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    const launchVSCodePage = new LaunchVSCodePage();
+    await new Promise(resolve => setTimeout(resolve, 20000)); // 20,000 milliseconds = 20 seconds
+    // await page.pause(); 
+
+    // Launch VSCode
+    await launchVSCodePage.launchVSCode(vscodeExecutablePath);
+
+    // Install the extension from the VSIX file
+    await launchVSCodePage.installExtensionFromVSIX(vsixFilePath);
+    await page.pause();
```
Comment (abrugaro) [inline e2e/tests/vscode.test.ts:25]: Remove this line as it will fail when we remove the page fixtura (it points to a browser, not to vscode)

File: e2e/tests/vscode.test.ts [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+    
+    // Path to the VSCode executable
+    const vscodeExecutablePath = '/usr/bin/code';  // Change this to the actual path
+
+    // Path to the VSIX file
+    const vsixFilePath = path.resolve(__dirname, '/home/sshveta/Downloads/kai-vscode-plugin-0.0.3.vsix');  // Provide the path to your VSIX file
+
+    // Create a new instance of the LaunchVSCodePage
+    const launchVSCodePage = new LaunchVSCodePage();
+    await new Promise(resolve => setTimeout(resolve, 20000)); // 20,000 milliseconds = 20 seconds
```
Comment (abrugaro) [inline e2e/tests/vscode.test.ts:17]: *question:* Why is this needed?

**Review by abrugaro — CHANGES_REQUESTED** [review]

**Review by abrugaro — CHANGES_REQUESTED** [review]

File: e2e/pages/vscode.pages.ts [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+    const moreActionsButton = await extensionsActionsButton.getByLabel(
+      "Views and More Actions...",
+    );
+    await moreActionsButton.click();
+
+    await this.window.click(
+      '.context-view .menu-item span:has-text("Install from VSIX...")',
+    );
+    // Set the file input for the VSIX file
+    const inputSelector = 'input[type="file"]';
```
Comment (midays) [inline e2e/pages/vscode.pages.ts:51]: for the selectors we can add them in a different file, `vscode.view.ts` maybe ?

File: e2e/pages/vscode.pages.ts [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+    await this.window.setInputFiles(inputSelector, vsixFilePath);
+
+    // Wait for the uninstall button to appear indicating the extension is installed
+    await this.window.waitForSelector(".extension-list-item .uninstall", {
+      timeout: 60000,
+    });
+  }
+
+  // Method to close VSCode
+  public async closeVSCode(): Promise<void> {
```
Comment (midays) [inline e2e/pages/vscode.pages.ts:61]: playwright closes vscode automatically after the tests finishes execting, I don't think there's a need for this.

**Review by midays — CHANGES_REQUESTED** [review]

**Review by abrugaro — APPROVED** [review]
Looks good for the first PR! We can improve it later!
